packages/demo-wallet-shared/methods/sep24/interactiveDepositFlow.ts

```diff
@@ -15,15 +28,19 @@ export const interactiveDepositFlow = async ({
   sep24TransferServerUrl,
   token,
   claimableBalanceSupported,
+  isCustodialMode,
 }: InteractiveDepositFlowProps) => {
   log.instruction({ title: "Starting SEP-24 interactive flow for deposit" });
 
   const formData = new FormData();
-  const postDepositParams = {
+  const postDepositParams: DepositParams = {
     asset_code: assetCode,
     account: publicKey,
     lang: "en",
     claimable_balance_supported: claimableBalanceSupported.toString(),
+    ...(isCustodialMode
+      ? { memo: Math.floor(Math.random() * 100).toString(), memo_type: "id" }
+      : {}),
```

Comment:
Ideally, we throw an error in the logs if the anchor does not use the memo generated here in the transaction that completes the deposit. Is that an easy lift?

Reply:
Or maybe not an error, but a warning?

Reply:
Added the check when we start polling transaction status. Here's an example of the warning.

![image](https://user-images.githubusercontent.com/7346473/170557734-56b9ed40-98bf-46aa-ade2-a25b5a4c8b0e.png)